packages/labs/signals/src/lib/signal-watcher.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import type {PropertyDeclaration, PropertyValueMap, ReactiveElement} from 'lit';
+import {Signal} from 'signal-polyfill';
+import {WatchDirective} from './watch.js';
+
+type ReactiveElementConstructor = abstract new (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ...args: any[]
+) => ReactiveElement;
+
+export interface SignalWatcher extends ReactiveElement {
+  updateWatch(d: WatchDirective<unknown>): void;
+}
+
+/**
+ * Adds the ability for a LitElement or other ReactiveElement class to
+ * watch for access to signals during the update lifecycle and trigger a new
+ * update when signals values change.
+ */
+export function SignalWatcher<T extends ReactiveElementConstructor>(
+  Base: T
+): T {
+  abstract class SignalWatcher extends Base {
+    // Watcher.watch() doesn't dedupe so we need to track this ourselves.
+    private __isWatching = false;
+    private __watcherUpdate = false;
+    private __watcher = new Signal.subtle.Watcher(() => {
+      this.__watcherUpdate = true;
+      this.requestUpdate();
+      this.__watcherUpdate = false;
+    });
+    private __forceUpdateSignal = new Signal.State(0);
+    private __updateSignal = new Signal.Computed(() => {
+      this.__forceUpdateSignal.get();
+      super.performUpdate();
```

Comment:
I think you have to call `this.__watcher.watch()` here to make sure to re-activate the watcher. Make sure to avoid passing the signal due to https://github.com/proposal-signals/proposal-signals/issues/157